src/extractor/extractor.cpp

```diff
@@ -194,6 +194,33 @@ int Extractor::run(ScriptingEnvironment &scripting_environment)
                     static_cast<const osmium::Node &>(*(osm_elements[result.first])),
                     result.second);
             }
+        }
+        nodeReader.close();
+
+        // Prepare cache data
+        extraction_containers.prepareCache();
+
+        // WAY AND RELATION PARSING
+        osmium::io::Reader wayReader(input_file, osmium::osm_entity_bits::way | osmium::osm_entity_bits::relation);
```

Comment:
same here, `read_metadata::no` needs to be passed

this flag alone accounts for ~10-20 performance while parsing nodes, ways, etc.